### PR TITLE
10 statistics

### LINKED ISSUE #10: Implement time-manager statistics
The current time-manager statistics (work hours left, earliest time to leave) are mocked and static.

Recalculate them dynamically from time-keeping data of the current day.

Ensure this works under day-to-day navigation.

### PR DESCRIPTION
**Pull request by danyball**
fixes #10

### CODE REVIEW

File: src/js/time-manager.js [hunk 1]
```diff
@@ -42,22 +43,54 @@ class TimeManager extends LitElement {
...
+    //init
+    this.workedHours = 0;
+    this.endTime = 0;
   }
 
-  getWorkedHours() {
-    return 0;
+  async setWorkedHours() {
+    const storedTimes = await getTimePairs('time-pairs-only');
+    const now = new Date();
```
Comment (markus-walther) [inline src/js/time-manager.js]: This and the next line are already contained in `human2decimalTime` of date-manipulation.js, when invoked with no argument. Just export that function there, and use it here (and below)!

File: src/js/time-manager.js [hunk 2]
```diff
@@ -42,22 +43,54 @@ class TimeManager extends LitElement {
...
+    const nowDecimal = timeToDecimal(now.getHours(), now.getMinutes());
+    this.workedHours = 0;
```
Comment (markus-walther) [inline src/js/time-manager.js]: Don't assign to a render-causing setter until your ultimate value is computed. => `let workedHours = 0;`

File: src/js/time-manager.js [hunk 3]
```diff
@@ -42,22 +43,54 @@ class TimeManager extends LitElement {
...
+        let endTime;
+
+        // no endTime is given and if given, the time is before now
+        if(isNaN(value[1]) && value[0] <= nowDecimal) {
+          //simulate endTime = now
+          endTime = nowDecimal;
+        } else {
+          endTime = value[1];
+        }
+        this.workedHours = this.workedHours + (endTime - value[0]);
```
Comment (markus-walther) [inline src/js/time-manager.js]: `workedHours += ...`

File: src/js/time-manager.js [hunk 4]
```diff
@@ -42,22 +43,54 @@ class TimeManager extends LitElement {
...
 
-  getWorkedHours() {
-    return 0;
+  async setWorkedHours() {
+    const storedTimes = await getTimePairs('time-pairs-only');
+    const now = new Date();
+    const nowDecimal = timeToDecimal(now.getHours(), now.getMinutes());
+    this.workedHours = 0;
+
+    storedTimes.forEach((value) => {
```
Comment (markus-walther) [inline src/js/time-manager.js]: value => `[startTime, stopTime]`, to get rid of ugly/non-DRY value[0/1] below

File: src/js/time-manager.js [hunk 5]
```diff
@@ -42,22 +43,54 @@ class TimeManager extends LitElement {
...
+      if(!isNaN(value[0])) {
```
Comment (markus-walther) [inline src/js/time-manager.js]: `isFinite` might be more readable than the double negation of ! and is-not-a-number.

File: src/js/time-manager.js [hunk 6]
```diff
@@ -42,22 +43,54 @@ class TimeManager extends LitElement {
...
+        let endTime;
+
+        // no endTime is given and if given, the time is before now
+        if(isNaN(value[1]) && value[0] <= nowDecimal) {
```
Comment (markus-walther) [inline src/js/time-manager.js]: `const noValidEndtime = isNaN(stopTime) && startTime <= nowDecimal`

File: src/js/time-manager.js [hunk 7]
```diff
@@ -42,22 +43,54 @@ class TimeManager extends LitElement {
...
+          //simulate endTime = now
+          endTime = nowDecimal;
+        } else {
+          endTime = value[1];
```
Comment (markus-walther) [inline src/js/time-manager.js]: not DRY. This is the perfect scenario for a ternary expression: `endTime = noValidEndtime ? nowDecimal : stopTime;`

File: src/js/time-manager.js [hunk 8]
```diff
@@ -42,22 +43,54 @@ class TimeManager extends LitElement {
...
+        this.workedHours = this.workedHours + (endTime - value[0]);
+      }
+    });
   }
 
-  getOvertime() {
-    return -1 * (this.totalHours - this.getWorkedHours());
+  setEndtime() {
+    dailyHours().then((hours) =>{
+      const now = new Date();
```
Comment (markus-walther) [inline src/js/time-manager.js]: see above

**Review by markus-walther — CHANGES_REQUESTED** [review]

File: index.js [hunk 9]
```diff
@@ -100,7 +100,14 @@ class TimeTracker extends LitElement {
     `;
   }
 
+  _triggerTimeManagerRefresh() {
+    let tmpDate = new Date(this.date);
+    tmpDate.setSeconds(tmpDate.getSeconds() + 1);
```
Comment (raphaellueckl) [inline index.js]: big hack!

**Pull request merged**